### PR TITLE
Remove AcceptedTermsOfServiceId from Users table

### PR DESCRIPTION
**Pull request by isacikgoz**
#### Summary
Adds a missing migration for users table.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-40778

#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: store/sqlstore/upgrade.go [hunk 1]
```diff
@@ -1429,3 +1431,13 @@ func upgradeDatabaseToVersion630(sqlStore *SqlStore) {
...
 	}
 }
+
+func upgradeDatabaseToVersion640(sqlStore *SqlStore) {
+	// if shouldPerformUpgrade(sqlStore, Version630, Version640) {
+	if sqlStore.DoesColumnExist("Users", "AcceptedTermsOfServiceId") {
+		sqlStore.GetMaster().ExecNoTimeout("ALTER TABLE Users DROP COLUMN AcceptedTermsOfServiceId")
+	}
+
+	// 	saveSchemaVersion(sqlStore, Version630)
```
Comment (streamer45) [inline store/sqlstore/upgrade.go:1441]:
```suggestion
	// 	saveSchemaVersion(sqlStore, Version640)
```

**Review by streamer45 — APPROVED** [review]
Thanks, looks good 👍  We may also consider this for a dot release in ESR in which case it will need its own 63x method. /cc @amyblais

**Comment by amyblais** [comment]
@isacikgoz I agree with including this for v6.3 as needed. I'll think about other versions.

**Comment by agnivade** [comment]
@amyblais - I am wondering if there's a reason we didn't backport this to 6.2 (the version where it first appeared)? A customer is running into this https://community-daily.mattermost.com/core/pl/3xuriogybirgjq5z4znjkc7bfw.

**Comment by amyblais** [comment]
> @amyblais - I am wondering if there's a reason we didn't backport this to 6.2 (the version where it first appeared)? A customer is running into this https://community-daily.mattermost.com/core/pl/3xuriogybirgjq5z4znjkc7bfw.

I asked Support at the time and they said that cherry-picking to 6.3 was fine since there was a simple workaround to fix it (it was a different customer) https://community-daily.mattermost.com/core/pl/xbmquz8oa7raxm3dz5mhxt3qfw. I'll follow the new thread and this can be backported as needed.

**Pull request merged**